tools/functional-tester/etcd-agent/agent.go

```diff
@@ -80,18 +82,40 @@ func (a *Agent) stop() error {
 	if a.state != stateStarted {
 		return nil
 	}
-	err := a.cmd.Process.Kill()
+
+	err := waitCmdExit(a.cmd)
 	if err != nil {
 		return err
 	}
-	_, err = a.cmd.Process.Wait()
+
+	err = a.cmd.Process.Kill()
+	if err.Error() == osutil.ErrProcessAlreadyFinished.Error() {
+		a.state = stateStopped
+		err = nil
+	}
+	return err
+}
+
+func waitCmdExit(cmd *exec.Cmd) error {
```

Comment:
After second thought, probably SigtermAndWait is a better naming?

/cc @heyitsanthony